components/scream/src/physics/p3/CMakeLists.txt

```diff
@@ -8,8 +8,8 @@ set(P3_SRCS
   atmosphere_microphysics_run.cpp
 )
 
-# Add ETI source files if not on CUDA
-if (NOT CUDA_BUILD OR Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
+# Add ETI source files if not on CUDA/HIP
+if (NOT EAMXX_ENABLE_GPU)
```

Comment:
Here I believe you should add back in `OR Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE`.

Reply:
In addition, below I see a reference to `KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE`; thus, we want `OR Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE OR Kokkos_ENABLE_HIP_RELOCATABLE_DEVICE_CODE`.

Even better would be to define `EAMXX_ENABLE_GPU_RELOCATABLE_DEVICE_CODE` as CMake and `#cmakedefine` variables.

Reply:
To be fair, I don't think we currently support a RDC build. I think I might have added this when I was trying to have our debug builds use rdc (and cut down the 5m compilation time for `[p3|shoc]_functions_f90.cpp`), but IIRC I hit an ICE in homme's bex files on CUDA. I might try to bring it back at some point...

Reply:
thanks, done